### PR TITLE
Query

### PR DESCRIPTION
**Pull request by simontegg**
woohooo!

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -7,14 +7,15 @@ var config = require('config')
...
+var Actions = require('actions')
 
 global.types = require('types')
 
-if (process.env.DEBUG) {
-  global.localStorage.setItem('debug', process.env.DEBUG)
-} else {
-  global.localStorage.removeItem('debug')
-}
+// if (process.env.DEBUG) {
```
Comment (ahdinosaur) [inline src/client.js]: i'm not sure if we should permanently disable debugging.

File: src/node_modules/api/index.js [hunk 2]
```diff
@@ -93,6 +94,22 @@ function createApi (config) {
           attrs.relationship = embeddedRoleMatch[1]
         }
 
+
+        //HACK for embedded arrays
```
Comment (ahdinosaur) [inline src/node_modules/api/index.js]: probably want to remove this?

File: src/node_modules/graph-view/get-default-query-map.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+var debug = require('debug')('graph-view:get-default-query-map')
+var parseQueryToMap = require('./parse-query-to-map')
+
+module.exports = function (model) {
+  debug('get-default-query-map', model)
+  var query = { q: [] }
+  if (model['@type'] === 'Group') {
+    if (model.childGroups.length > 0) {
+      query.q.push({
+        link: '/linkTypes/has-subgroup',
```
Comment (ahdinosaur) [inline src/node_modules/graph-view/get-default-query-map.js:10]: leaving a mental note that this is where we hardcode link types.

File: src/node_modules/graph-view/get-queries.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    'type': 'Query',
+    component: options.component,
+    id: options.id,
+    source: options.source,
+    target: options.target,
+    context: options.context,
+    labels: makeLabels(options),
+    showHalo: options.showHalo,
+    showLink: options.showLink,
+    isParent: options.component.id === '/linkTypes/is-part-of',
```
Comment (ahdinosaur) [inline src/node_modules/graph-view/get-queries.js:86]: more hard-coding

**Comment by ahdinosaur** [comment]
legit as. :cake: congrats on more line deletions than additions. :wink:

**Comment by simontegg** [comment]
ping @ahdinosaur

**Comment by ahdinosaur** [comment]
sorry for the delay, i'll try and deploy this and https://github.com/holodex/enspiral-data/pull/21/files as soon as possible.

**Comment by ahdinosaur** [comment]
[deployed!](http://holodex.enspiral.info/groups/enspiral-foundation) :tada:

**Pull request merged**